### PR TITLE
[REACTOS] Fix GetTokenInformation() usage

### PR DESCRIPTION
**Pull request by SergeGautherie**
The first call to GetTokenInformation is used to determine the size of a TokenInformation buffer.
It should fail and return ERROR_INSUFFICIENT_BUFFER

Follow-up to 98bbe83.

### CODE REVIEW

File: base/system/winlogon/sas.c [hunk 1]
```diff
@@ -432,27 +432,19 @@ AllowWinstaAccess(PWLSESSION Session)
...
     // Get required buffer size and allocate the TOKEN_GROUPS buffer.
-
-    if (!GetTokenInformation(Session->UserToken,
-                             TokenGroups,
-                             ptg,
-                             0,
-                             &dwLength))
+    if (GetTokenInformation(Session->UserToken, TokenGroups, NULL, 0, &dwLength) ||
+        GetLastError() != ERROR_INSUFFICIENT_BUFFER ||
+        !(ptg = (PTOKEN_GROUPS)HeapAlloc(GetProcessHeap(), HEAP_ZERO_MEMORY, dwLength)))
```
Comment (HBelusca) [inline base/system/winlogon/sas.c:445]: Please, don't do that.

File: modules/rostests/tests/tokentest/tokentest.c [hunk 2]
```diff
@@ -278,18 +278,20 @@ GetFromToken(HANDLE hToken, TOKEN_INFORMATION_CLASS tic)
...
     DWORD n;
-	PBYTE p = 0;
+	PBYTE p;
 
-    bResult = GetTokenInformation(hToken, tic, 0, 0, &n);
-    if ( ! bResult && GetLastError() != ERROR_INSUFFICIENT_BUFFER)
-		return 0;
+    bResult = GetTokenInformation(hToken, tic, NULL, 0, &n);
+    if (bResult || GetLastError() != ERROR_INSUFFICIENT_BUFFER ||
+        !(p = (PBYTE) malloc(n)))
```
Comment (HBelusca) [inline modules/rostests/tests/tokentest/tokentest.c:285]: Ugly

File: base/applications/cmdutils/eventcreate/eventcreate.c [hunk 3]
```diff
@@ -126,8 +126,9 @@ GetUserToken(
     BOOL Success = FALSE;
     DWORD dwError;
     HANDLE hToken;
-    DWORD cbTokenBuffer = 0;
+    DWORD cbTokenBuffer;
     PTOKEN_USER pUserToken = NULL;
+    BOOL bRet;
```
Comment (HBelusca) [inline base/applications/cmdutils/eventcreate/eventcreate.c:131]: There is already the "Success" variable for this sort of things.

**Comment by SergeGautherie** [comment]
> There is already the "Success" variable for this sort of things.

`Success` is the return value of the function. I prefer not to mess with it.

**Comment by GeoB99** [comment]
> 
> 
> > There is already the "Success" variable for this sort of things.
> 
> `Success` is the return value of the function. I prefer not to mess with it.

This doesn't make sense. Unless you've a legitimate reason to have a separate return boolean variable, as it stands now, it just wastes stack space for no gain. As long as you know what you're doing, you can't mess up.

**Comment by GeoB99** [comment]
Closing this for being a stale PR. It'll be opened when the author continues working on this.

**Pull request closed without merging**